README.md

```diff
@@ -1,2 +1,84 @@
 # next-webpack-oneof
-Wraps all weabpack loader rules into a single oneOf rule.
+
+[![NPM version][npm-image]][npm-url] [![Downloads][downloads-image]][npm-url] [![Build Status][travis-image]][travis-url] [![Coverage Status][codecov-image]][codecov-url] [![Dependency status][david-dm-image]][david-dm-url] [![Dev Dependency status][david-dm-dev-image]][david-dm-dev-url]
+
+[npm-url]:https://npmjs.org/package/@moxy/next-webpack-oneof
+[downloads-image]:https://img.shields.io/npm/dm/@moxy/next-webpack-oneof.svg
+[npm-image]:https://img.shields.io/npm/v/@moxy/next-webpack-oneof.svg
+[travis-url]:https://travis-ci.org/moxystudio/next-webpack-oneof
+[travis-image]:http://img.shields.io/travis/moxystudio/next-webpack-oneof/master.svg
+[codecov-url]:https://codecov.io/gh/moxystudio/next-webpack-oneof
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/next-webpack-oneof/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/next-webpack-oneof
+[david-dm-image]:https://img.shields.io/david/moxystudio/next-webpack-oneof.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/next-webpack-oneof?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/next-webpack-oneof.svg
+
+Wraps all webpack loader rules into a single oneOf rule.
+
+By default webpack loaders will test all files for all loaders. However, webpack does support [`oneOf` rules](https://webpack.js.org/configuration/module/#ruleoneof), within which only the first matching rule will be used.
+
+This plugin transfers all rules into a single `oneOf` rule, so no file will match more than one condition.
+
+
+## Installation
+
+```sh
+$ npm install --save @moxy/next-webpack-oneof
+```
+
+
+## Usage
+
+In your `next.config.js` file:
+
+```js
+const alwaysOneOf = require('@moxy/next-webpack-oneof');
+
+module.exports = alwaysOneOf({...nextConfig});
+```
+
+This plugin must be passed as the first loader rule in your `next.config.js` file, as it works by spreading rules already present in your webpack `config` into a `oneOf` object, and must be placed where it will have access to all rules, which, as per the webpack implementation (explained further below), is **before** the rest of your rules in your exporting object.
```

Comment:
Clarify that it must be the last one to be executed, meaning it must be the first one in the compose chain.

Reply:
Add a period after `into a `oneOf` object, ` and start the next sentence with something like `Therefore`

Reply:
> which, as per the webpack implementation

should be "as per the way you compose plugins in Next.js"